src/parsers/manifest/dash/manifest_bounds_calculator.ts

```diff
@@ -0,0 +1,146 @@
+ *
+ * @example
+ * ```js
+ * const manifestBoundsCalculator = new ManifestBoundsCalculator();
+ *
+ * // let's imagine a property `index` on a Representation which need to obtain
+ * // the last position at any time
+ * someRepresentation.index = new TemplateRepresentationIndex({
+ *   // ...
+ *   manifestBoundsCalculator // for now, `getLastPosition` will return `undefined`
+ * });
+ *
+ * // ...
+ * // Let's imagine a function which try to guess the last position based on a
+ * // given parsed period
+ * const lastPosition = getLastPositionForPeriod(somePeriod);
+ * if (lastPosition != null) {
+ *   const clockOffset = performance.now() / 1000;
+ *   manifestBoundsCalculator.setLastPosition(lastPosition, clockOffset);
+ *   // `getLastPosition` will now be correctly communicate the last position
+ *   // (it returned `undefined` until then).
+ * }
+ * ```
+ * @class ManifestBoundsCalculator
+ */
+export default class ManifestBoundsCalculator {
+  private _timeShiftBufferDepth : number | null;
+  private _availabilityStartTime : number;
+  private _clockOffset : number | undefined;
+  private _lastPosition : number | undefined;
+  private _firstPosition : number | undefined;
+  private _isDynamic : boolean;
+
+  constructor(args : { timeShiftBufferDepth? : number;
+                       availabilityStartTime : number;
+                       isDynamic : boolean; }
+  ) {
+    this._isDynamic = args.isDynamic;
+    this._firstPosition = !args.isDynamic ? 0 : undefined;
+    this._timeShiftBufferDepth = !args.isDynamic ||
+                                 args.timeShiftBufferDepth == null ?
+                                   null :
+                                   args.timeShiftBufferDepth;
+    this._availabilityStartTime = args.availabilityStartTime;
+  }
+
+  /**
+   * Set the last position and the clock offset (the value of `performance.now()`
+   * at the time that position was true converted into seconds).
+   *
+   * @example
+   * Example if you trust `Date.now()` to give you a reliable offset:
+   * ```js
+   * const lastPosition = Date.now();
+   * const clockOffset = performance.now() / 1000;
+   * manifestBoundsCalculator.setLastPosition(lastPosition, clockOffset);
+   * ```
+   *
+   * @param {number} lastPositionOffset
+   */
+  setLastPosition(lastPosition : number, clockOffset?: number) {
+    this._lastPosition = lastPosition;
+    this._clockOffset = clockOffset;
+  }
+
+  /**
+   * Returns `true` if the last position and the clock offset (for live content only)
+   * have been comunicated.
+   * `false` otherwise.
+   * @returns {boolean}
+   */
+  lastPositionIsKnown() : boolean {
+    if (this._isDynamic) {
+      return this._clockOffset != null && this._lastPosition != null;
+    }
+    return this._lastPosition != null;
+  }
+
+  /**
+   * Get first available position.
+   * @return {number|undefined}
+   */
+  getFirstAvailablePosition(): number | undefined {
```

Comment:
Isn't it more a `minimumBound`? First available position would let us think that that position is available